observability-kit-agent/src/main/java/com/vaadin/extension/instrumentation/client/ObjectMapSpanData.java

```diff
@@ -0,0 +1,278 @@
+            EventData eventData = EventData.create(
+                    (long) event.get("timeUnixNano"),
+                    (String) event.get("name"),
+                    getAttributes(event).build());
+            events.add(eventData);
+        }
+
+        this.links = Collections.emptyList();
+
+        Map<String, Object> status = (Map<String, Object>) span.get("status");
+        this.status = ((int) status.get("code") == 0) ? StatusData.ok() :
+                StatusData.error();
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private static AttributesBuilder getAttributes(Map<String, Object> node) {
+        AttributesBuilder builder = Attributes.builder();
+        if (node.containsKey("attributes")) {
+            List<Map<String, Object>> attributes =
+                    (List<Map<String, Object>>) node.get("attributes");
+            for (Map<String, Object> attribute : attributes) {
+                String key = (String) attribute.get("key");
+                Map<String, Object> value =
+                        (Map<String, Object>) attribute.get("value");
+                Map.Entry<AttributeKey, Object> entry =
+                        attributeValue(key, value);
+                if (entry != null) {
+                    builder.put(entry.getKey(), entry.getValue());
+                }
+            }
+        }
+        return builder;
+    }
+
+    // https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_otlp_transformer.IAnyValue.html
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private static Map.Entry<AttributeKey, Object> attributeValue(
+            String key, Map<String, Object> value) {
+        if (value.containsKey("stringValue")) {
+            String stringValue = (String) value.get("stringValue");
+            return new AbstractMap.SimpleImmutableEntry<>(
+                    AttributeKey.stringKey(key), stringValue);
+        } else if (value.containsKey("intValue")) {
+            Integer intValue = (Integer) value.get("intValue");
+            long longValue = (long) intValue;
+            return new AbstractMap.SimpleImmutableEntry<>(
+                    AttributeKey.longKey(key), longValue);
+        } else if (value.containsKey("boolValue")) {
+            Boolean boolValue = (Boolean) value.get("boolValue");
+            return new AbstractMap.SimpleImmutableEntry<>(
+                    AttributeKey.booleanKey(key), boolValue);
+        } else if (value.containsKey("doubleValue")) {
+            Double doubleValue = (Double) value.get("doubleValue");
+            return new AbstractMap.SimpleImmutableEntry<>(
+                    AttributeKey.doubleKey(key), doubleValue);
+        } else if (value.containsKey("arrayValue")) {
+            AttributeKey<?> attributeKey = null;
+            List<Object> values = new ArrayList<>();
+            for (Map<String, Object> item :
+                    (List<Map<String, Object>>) value.get("arrayValue")) {
```

Comment:
Since we need the cast to List only to iterate, what about casting to `Collection` instead?